### PR TITLE
Correct the OUTPUT of add_custom_command so CMake/Ninja know when it doesn't need to run

### PR DESCRIPTION
**Pull request by afuller-TT**
### Issue
#391 

### Description
ninja && sleep 1 && ninja will cause the header to be generated a second time, with a new timestamp, and downstream will re-link binaries.

### List of the changes
Fixed the OUTPUT of the custom command to the correct file (which is derived from the basename of the file, not the name+ext).  Now CMake/Ninja know when to NOT regenerate the header.

### Testing
Ran compiles locally.

### API Changes
None.

### CODE REVIEW

File: device/CMakeLists.txt [hunk 1]
```diff
@@ -1,15 +1,13 @@
 set(POSITION_INDEPENDENT_CODE ON)
 
 set(FBS_FILE ${PROJECT_SOURCE_DIR}/device/simulation/tt_simulation_device.fbs)
-get_filename_component(FBS_FILE_NAME ${FBS_FILE} NAME)
+get_filename_component(FBS_FILE_NAME ${FBS_FILE} NAME_WLE)
 set(FBS_GENERATED_HEADER "${CMAKE_CURRENT_BINARY_DIR}/${FBS_FILE_NAME}_generated.h")
```
Comment (afuller-TT) [inline device/CMakeLists.txt:5]: This filename is `tt_simulation_device_generated.h`, not `tt_simulation_device.fbs_generated.h`.  So updated the line above to ask for without the last ext (WLE).

Comment (afuller-TT) [reply device/CMakeLists.txt:5]: For reference this cuts down on 148 targets in tt-metal being needlessly re-build/linked.

**Review by blozano-tt — APPROVED** [review]
LGTM

File: device/CMakeLists.txt [hunk 2]
```diff
@@ -1,15 +1,13 @@
...
 add_custom_command(
     OUTPUT
         ${FBS_GENERATED_HEADER}
     COMMAND
-        flatc
-    ARGS
```
[the remark this answers is not in the document]
Comment (broskoTT) [reply device/CMakeLists.txt:11]: Last time, mingling with this flatc commang broke clang-tidy in all-post-commit. Can you verify this won't break by running all post commit?

[the remark this answers is not in the document]
Comment (broskoTT) [reply device/CMakeLists.txt:11]: Ah nvm, you just moved it all under COMMAND. I didn't read what was the diff when I gave last comment. No action

**Review by broskoTT — APPROVED** [review]

**Pull request merged**